src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -4,260 +4,112 @@ open Equinox.Core
+            let withRetryTag count =
+                Activity.setTags [|Tags.retries, count|]
+                f
+            retryPolicy withRetryTag
 
 module private Write =
 
-    let private writeEventsAsync (writer: MessageDbWriter) streamName version events ct: Task<MdbSyncResult> =
-        writer.WriteMessages(streamName, events, version, ct)
     let inline len (bytes: EventBody) = bytes.Length
     let private eventDataLen (x: IEventData<EventBody>) = len x.Data + len x.Meta
     let private eventDataBytes events = events |> Array.sumBy eventDataLen
-    let private writeEventsLogged writer streamName version events (log: ILogger) ct: Task<MdbSyncResult> = task {
-        let act = Activity.Current
-        let log = if not (log.IsEnabled Events.LogEventLevel.Debug) then log else log |> Log.propEventData "Json" events
-        let bytes, count = eventDataBytes events, events.Length
-        let log = log |> Log.prop "bytes" bytes
-        if act <> null then act.AddExpectedVersion(version).IncMetric(count, bytes) |> ignore
-        let! t, result = writeEventsAsync writer streamName version events |> Stopwatch.time ct
-        let reqMetric: Log.Measurement = { stream = streamName; interval = t; bytes = bytes; count = count}
-        let resultLog, evt =
-            match result with
-            | MdbSyncResult.Written x ->
-                if act <> null then
-                    act.SetStatus(ActivityStatusCode.Ok).AddTag("eqx.new_version", x) |> ignore
-                log |> Log.prop "currentPosition" x, Log.WriteSuccess reqMetric
-            | MdbSyncResult.ConflictUnknown ->
-                let eventTypes = [| for x in events -> x.EventType |]
-                if act <> null then act.RecordConflict().AddTag("eqx.event_types", eventTypes) |> ignore
-                let writeLog = log |> Log.prop "stream" streamName |> Log.prop "count" count
-                writeLog.Information("MdbTrySync WrongExpectedVersionException writing {eventTypes}, expected {expectedVersion}", eventTypes, version)
-                log, Log.WriteConflict reqMetric
-        (resultLog |> Log.event evt).Information("Mdb{action:l} count={count} conflict={conflict}",
-                                                 "Write", count, match evt with Log.WriteConflict _ -> true | _ -> false)
+    let private writeEventsLogged (writer: MessageDbWriter) streamName version events ct: Task<MdbSyncResult> = task {
+        Activity.setTags [|Tags.append_bytes, eventDataBytes events|]
+        let! result = writer.WriteMessages(streamName, events, version, ct)
+        match result with
+        | MdbSyncResult.Written x -> Activity.setTags [|Tags.new_version, x|]
+        | MdbSyncResult.ConflictUnknown ->
+            let eventTypes =
+                if events.Length <= 3
+                then [| for x in events -> x.EventType |]
+                else [| for x in Seq.take 3 events -> x.EventType |]
```

Comment:
if not doing a Seq.distinct, then use an unconditional Seq.truncate instead of the condition